sp/include/optee_sp_internal_api.h

```diff
@@ -0,0 +1,67 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2020-2021, Arm Limited.
+ */
+
+#ifndef SP_INCLUDE_OPTEE_SP_INTERNAL_API_H_
+#define SP_INCLUDE_OPTEE_SP_INTERNAL_API_H_
+
+#include <stddef.h>
+#include <stdint.h>
+#include "compiler.h"
+#include "util.h"
+
+/*
+ * The file describes the API between the SP dev kit and the SP. It contains the
+ * function prototypes that the user code must define.
+ * The SP code base should also contain a header file named
+ * "optee_sp_user_defines.h" for passing the following definitions to the SP dev
+ * kit:
+ * * OPTEE_SP_HEAP_SIZE: Heap size in bytes
+ * * OPTEE_SP_UUID: UUID of the SP as an sp_uuid structure
+ * * OPTEE_SP_STACK_SIZE: Stack size in bytes
+ * * OPTEE_SP_FLAGS: SP attributes (currently none available, set to zero)
+ */
+
+/*
+ * The file defines the functions and variables that other
+ */
+
+/*
+ * SP header types
+ */
+struct optee_sp_uuid {
+	uint32_t timeLow;
+	uint16_t timeMid;
+	uint16_t timeHiAndVersion;
+	uint8_t clockSeqAndNode[8];
+};
+
+struct optee_sp_head {
+	struct optee_sp_uuid uuid;
+	uint32_t stack_size;
+	uint32_t flags;
+	uint64_t depr_entry;
```

Comment:
I don't think you need this deprecated field from `struct ta_head`.

Reply:
Currently there's an implicit dependency between `struct ta_head` and `struct sp_head` as [ldelf always uses](https://github.com/OP-TEE/optee_os/blob/master/ldelf/ta_elf.c#L1073) `struct ta_head` regardless of if it's loading a TA or an SP. So essentially the two structs should be the same.
One solution would be to pass a parameter to ldelf so it can diverge according to the type (TA or SP) of the entity being loaded.
However the preferred way of loading SPs in SPMs is to avoid ELF loader and simply have the SPs as raw binaries. So in the near future we are planning to omit ldelf from the SP load process so I don't think it's worth the effort of modifying the ldelf. As far as I know @jenswi-linaro is aware of these discussions.
If this sounds reasonable I would leave it this way for now. I can also add a comment of this explanation to the code.

Reply:
> Then why not use struct ta_head for now?
`struct ta_head` is defined in `user_ta_header.h` of libutee and it depends on other files of libutee. I didn't want to involve libutee in the build process of SPs as they wouldn't use much of it.

> Funny thing, we went basically the opposite way for TAs... but I do understand the use cases are a bit different.
Yes, I guess it's mostly because of other SPM solutions use the raw binary format so it is better the have OP-TEE aligned with this method as well.

Reply:
minor: maybe rename `depr_entry` to `reserved`.

Reply:
Done